collect_app/src/test/java/org/odk/collect/android/preferences/stores/AdminPreferencesDataStoreTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.odk.collect.android.preferences.stores
+
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.odk.collect.android.preferences.PreferencesDataSource
+import org.odk.collect.android.preferences.PreferencesDataSourceProvider
+
+class AdminPreferencesDataStoreTest {
+    private lateinit var adminPreferencesDataStore: AdminPreferencesDataStore
+    private lateinit var preferencesDataSourceProvider: PreferencesDataSourceProvider
+    private lateinit var adminPreferences: PreferencesDataSource
+
+    @Before
+    fun setup() {
+        preferencesDataSourceProvider = mock(PreferencesDataSourceProvider::class.java)
+        adminPreferences = mock(PreferencesDataSource::class.java)
+        `when`(preferencesDataSourceProvider.getAdminPreferences()).thenReturn(adminPreferences)
+        adminPreferencesDataStore = AdminPreferencesDataStore(preferencesDataSourceProvider)
+    }
+
+    @Test
+    fun `When putString() called, should save() be called on adminPreferences`() {
+        adminPreferencesDataStore.putString("Key", "value")
+        verify(adminPreferences, times(1)).save("Key", "value")
+    }
+
+    @Test
+    fun `When putBoolean() called, should save() be called on adminPreferences`() {
+        adminPreferencesDataStore.putBoolean("Key", true)
+        verify(adminPreferences, times(1)).save("Key", true)
+    }
+
+    @Test
+    fun `When putLong() called, should save() be called on adminPreferences`() {
+        adminPreferencesDataStore.putLong("Key", 3L)
+        verify(adminPreferences, times(1)).save("Key", 3L)
+    }
+
+    @Test
+    fun `When putInt() called, should save() be called on adminPreferences`() {
+        adminPreferencesDataStore.putInt("Key", 5)
+        verify(adminPreferences, times(1)).save("Key", 5)
+    }
+
+    @Test
+    fun `When putFloat() called, should save() be called on adminPreferences`() {
+        adminPreferencesDataStore.putFloat("Key", 8.43f)
+        verify(adminPreferences, times(1)).save("Key", 8.43f)
+    }
+
+    @Test
+    fun `When putStringSet() called, should save() be called on adminPreferences`() {
+        adminPreferencesDataStore.putStringSet("Key", setOf("x", "y", "z"))
+        verify(adminPreferences, times(1)).save("Key", setOf("x", "y", "z"))
+    }
+
+    @Test
+    fun `When getString() called, should getString() be called on adminPreferences`() {
+        adminPreferencesDataStore.getString("Key", "value")
+        verify(adminPreferences, times(1)).getString("Key")
```

Comment:
This doesn't actually verify that we are returning the right thing from `AdminPreferencesDataStore#getString`. We either use `when` to set up the return values and assert they come back or just create a fake (probably backed by a map) `PreferenceDataSource` implementation that we can add values to.  I think for simple cases like this the fake might be easier.

Reply:
Ok I improved that.